### PR TITLE
MCLOUD-6660: Add possibility to disable MailHog

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description

Added possibility to disable MailHog.
Removed unused code.

- https://jira.corp.magento.com/browse/MCLOUD-6660
- https://github.com/magento/magento-cloud-docker/issues/271

### Manual testing scenarios

1. run `vendor/bin/ece-docker build:compose  --no-mailhog` command
2. Open `docker-compose.yaml` check that `mailhog` service is absent
3. run `vendor/bin/ece-docker build:compose` command
4. Open `docker-compose.yaml` check that `mailhog` service is present
### Release notes

Added the `--no-mailhog` option to the `build:compose` command to disable the MailHog service.

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Config/Source/CliSource.php [hunk 1]
```diff
@@ -166,11 +168,9 @@ public function read(): Repository
         }
 
         foreach (self::$disableOptionsMap as $option => $service) {
-            if ($value = $this->input->getOption($option)) {
-                $repository->set([
-                    $service . '.enabled' => false
-                ]);
-            }
+            $repository->set([
+                $service . '.enabled' => !$this->input->getOption($option)
```
Comment (shiftedreality) [inline src/Config/Source/CliSource.php]: It should be the same as old code

Comment (shiftedreality) [reply src/Config/Source/CliSource.php]: Move the default enabled value to `CloudSource`

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Config/Source/CliSource.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Source/CliSource.php]: Moved

**Review by shiftedreality — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**